src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -205,15 +206,16 @@ private JsonConverter GetConverterFromAttribute(JsonConverterAttribute converter
             }
             else
             {
-                ConstructorInfo ctor = type.GetConstructor(Type.EmptyTypes);
-                if (!typeof(JsonConverter).IsAssignableFrom(type) || !ctor.IsPublic)
+                ConstructorInfo? ctor = type.GetConstructor(Type.EmptyTypes);
+                if (!typeof(JsonConverter).IsAssignableFrom(type) || ctor == null || !ctor.IsPublic)
                 {
                     ThrowHelper.ThrowInvalidOperationException_SerializationConverterOnAttributeInvalid(classTypeAttributeIsOn, propertyInfo);
                 }
 
-                converter = (JsonConverter)Activator.CreateInstance(type);
+                converter = (JsonConverter?)Activator.CreateInstance(type);
```

Comment:
When will `Activator.CreateInstance` return null?

Reply:
cc @GrabYourPitchforks, @steveharter

Reply:
> When will Activator.CreateInstance return null?

I heard it will return null for Nullable value type

Reply:
> I heard it will return null for Nullable value type

In that case, given we already have the check on `type` above that `JsonConverter` is assignable from it, I don't think we need to worry about `CreateInstance` returning null here.

Reply:
Also row on 201 `converter = converterAttribute.CreateConverter(typeToConvert);` could return nullable. Well that is virtual method just returning null, but as there were no implementation I marked it nullable

Reply:
Yes, that's fine. Keep the local nullable (i.e. `JsonConverter? converter;`). Just change this particular line:
```suggestion
                converter = (JsonConverter)Activator.CreateInstance(type);
```

Reply:
well then we will need to add bang after calling `Activator.CreateInstance(type)` because the method return nullable but we are casting if to non null `JsonConverter`, we normally just add ? for casting type

Reply:
> But am fine with banging too

Same. Go for it.